### PR TITLE
Created Community GET Endpoint

### LINKED ISSUE #49: Community GET endpoint
Server and Client

Server Side:
We need a way to retrieve the different communities stored in our database.
We have to create a new route for communities and make endpoints that retrieve them.

We will need at least 1 GET endpoint:
- 1 to get all communities 

Client Side:
We need a way to call this endpoint. This will be done through the redux toolkit query store implementation setup (RTK Query).
Look at user.slice in the store for inspiration.

### PR DESCRIPTION
**Pull request by KingJordan152**
## Description

In this PR, I created a GET endpoint for acquiring all the communities currently stored in the database. This will be extremely helpful later on as we develop our Community page (#110). 

## Related Issue
This PR closes #49 and should be used in #110. 

## Type of Changes
Please select the relevant option:
- [ ] Bug fix
- [x] New feature
- [x] Enhancement
- [ ] Documentation update
- [ ] Other (please specify)
      
### Changes

- Created a route in Express for getting all the communities: `/api/community/all-communities`.
- Created a controller to handle querying the database for all communities.
- Created a slice that can be used to the call the endpoint on the client.
- Created type interfaces for all our models so our response types are more accurate.

## Things to Update
Please add any features, buttons, or pages that might need updating in the future:
- [ ] Button Functunality 
- [ ] Code Documentation 
- [ ] Mobile Responsiveness 
- [ ] Code testing 
- [x] Other (please specify)

I wasn't able to test this thoroughly since we don't have a way to create communities yet. We might have to update the `route`, `controller`,  and `slice` files to include an endpoint that handles this soon.

If any difficulties with this code become present once we start using it, I'll quickly fix it.

## Screenshots 
[Insert any relevant screenshots or GIFs to showcase visual changes, if applicable]

**N/A**

## Additional Notes
[Provide any additional information or context for reviewers]

**N/A**

### CODE REVIEW

**Review by Aligary — APPROVED** [review]
Everything looks good. This will definitely be useful when working on the community page.

File: server/src/routes/Community.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import express from 'express';
+import * as communityController from '../controllers/CommunityController';
+import { validateToken } from '../middlewares';
+
+const router = express.Router();
+
+router.get(
+  '/all-communities',
```
Comment (Linkaloo) [inline server/src/routes/Community.ts:8]: This isn't a big deal just kind of a naming thing with restful apis. This route is already under "api/community" even though its singular in name, that route would actually provide a list of all of them. Meanwhile, a route like "api/community/{community_id}" would provide information of a specific community based on the id. This hits the 2 usual requirements of everything and something specific.

I've seen both singular and plural used for these base routes, "users" instead of "user". So if that seems more natural to have this pattern we can change to that in another pr or leave it as is.

TLDR: "api/communities/" would be a route that returns a list of all of them and "api/communities/{id}" would be specific, instead of having a route called "all-communities".

**Review by Linkaloo — APPROVED** [review]

**Review by dafneMena — APPROVED** [review]
Hi Jordan, great work. Thank you for implementing this - it will definitely be needed to move forward with the community page. 

No additional comments on my end, all looks good!

File: server/src/routes/Community.ts (hunk 1 above)
Comment (KingJordan152) [reply server/src/routes/Community.ts:8]: That makes a lot of sense, thank you for the tip. I think for the time being we can just leave this as-is and revisit it later.
